File: implementations/java-springboot/harness/test/testdata/soft-delete-filter/good-global-annotation/account/infrastructure/persistence/AccountJpaEntity.java
package com.example.accountservice.account.infrastructure.persistence;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import java.time.LocalDateTime;
import org.hibernate.annotations.SQLRestriction;

@Entity
@SQLRestriction("deleted_at IS NULL")
public class AccountJpaEntity {

    @Column private LocalDateTime deletedAt;

    public LocalDateTime getDeletedAt() {
        return deletedAt;
    }
}

File: implementations/kotlin-springboot/harness/test/testdata/soft-delete-filter/good-sql-restriction/src/main/kotlin/account/infrastructure/persistence/AccountJpaEntity.kt
package com.example.accountservice.account.infrastructure.persistence

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDateTime
import org.hibernate.annotations.SQLRestriction

@Entity
@Table(name = "accounts")
@SQLRestriction("deleted_at IS NULL")
class AccountJpaEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @Column
    var deletedAt: LocalDateTime? = null,
)
